libtiledbsoma/src/pyapi/libtiledbsoma.cc

```diff
@@ -283,24 +283,110 @@ PYBIND11_MODULE(libtiledbsoma, m) {
 
         .def(
             "set_dim_points",
+            static_cast<void (SOMAReader::*)(
+                const std::string&, const std::vector<int64_t>&)>(
+                &SOMAReader::set_dim_points))
+
+        .def(
+            "set_dim_points_string_or_bytes",
             static_cast<void (SOMAReader::*)(
                 const std::string&, const std::vector<std::string>&)>(
                 &SOMAReader::set_dim_points))
 
-        // Binding to set slices using PyArrow::ChunkedArray
         .def(
-            "set_dim_points",
+            "set_dim_points_float64",
+            static_cast<void (SOMAReader::*)(
+                const std::string&, const std::vector<double>&)>(
+                &SOMAReader::set_dim_points))
+
+        .def(
+            "set_dim_points_float32",
+            static_cast<void (SOMAReader::*)(
+                const std::string&, const std::vector<float>&)>(
+                &SOMAReader::set_dim_points))
+
+        .def(
+            "set_dim_points_int64",
+            static_cast<void (SOMAReader::*)(
+                const std::string&, const std::vector<int64_t>&)>(
+                &SOMAReader::set_dim_points))
+
+        .def(
+            "set_dim_points_int32",
+            static_cast<void (SOMAReader::*)(
+                const std::string&, const std::vector<int32_t>&)>(
+                &SOMAReader::set_dim_points))
+
+        .def(
+            "set_dim_points_int16",
+            static_cast<void (SOMAReader::*)(
+                const std::string&, const std::vector<int16_t>&)>(
+                &SOMAReader::set_dim_points))
+
+        .def(
+            "set_dim_points_int8",
+            static_cast<void (SOMAReader::*)(
+                const std::string&, const std::vector<int8_t>&)>(
+                &SOMAReader::set_dim_points))
+
+        .def(
+            "set_dim_points_uint64",
+            static_cast<void (SOMAReader::*)(
+                const std::string&, const std::vector<uint64_t>&)>(
+                &SOMAReader::set_dim_points))
+
+        .def(
+            "set_dim_points_uint32",
+            static_cast<void (SOMAReader::*)(
+                const std::string&, const std::vector<uint32_t>&)>(
+                &SOMAReader::set_dim_points))
+
+        .def(
+            "set_dim_points_uint16",
+            static_cast<void (SOMAReader::*)(
+                const std::string&, const std::vector<uint16_t>&)>(
+                &SOMAReader::set_dim_points))
+
+        .def(
+            "set_dim_points_uint8",
+            static_cast<void (SOMAReader::*)(
+                const std::string&, const std::vector<uint8_t>&)>(
+                &SOMAReader::set_dim_points))
+
+        // The above short functions are expected to be invoked when the coords
+        // are Python list/tuple, or NumPy arrays.  Arrow arrays are in this
+        // long if-else-if function.
+        .def(
+            "set_dim_points_arrow_array",
```

Comment:
Maybe call the function `set_dim_points_arrow` to be less specific? It could be an `Array` or a `ChunkedArray`.

Reply:
good call